### PR TITLE
Add action icon examples

### PR DESCRIPTION
**Pull request by itwasmattgregg**
### WHY are these changes introduced?

We want to encourage the use of icons only for commonly used buttons as stated in the Patterns section, and also encourage more use of icons in buttons with text to show what the verb of the button does.

### WHAT is this pull request doing?

This PR adds some extra examples and changes others to encourage use of icons in Page and Card actions.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Page/components/Header/Header.tsx [hunk 1]
```diff
@@ -309,14 +309,13 @@ function shouldShowIconOnly(
...
-  let {content, accessibilityLabel, icon} = action;
+  let {content, accessibilityLabel} = action;
+  const {icon} = action;
   if (icon == null) return {...action, icon: undefined};
 
   if (isMobile) {
     accessibilityLabel = accessibilityLabel || content;
     content = undefined;
-  } else {
-    icon = undefined;
```
Comment (alex-page) [inline polaris-react/src/components/Page/components/Header/Header.tsx]: Any context on why this change is included with example updates?

Comment (itwasmattgregg) [reply polaris-react/src/components/Page/components/Header/Header.tsx]: When trying to add an icon to the primary action of a Page component I discovered that this code actually prevents that. Should I make a separate PR for that?

Comment (alex-page) [reply polaris-react/src/components/Page/components/Header/Header.tsx]: Yes please because this is changing functionality of the component or fixing a bug.

File: polaris.shopify.com/pages/examples/page-full-width.tsx [hunk 2]
```diff
@@ -8,15 +8,26 @@ function PageExample() {
...
-      <LegacyCard title="Credit card" sectioned>
-        <p>Credit card information</p>
-      </LegacyCard>
+      <Card>
+        <Text as="h2" variant="headingSm">
+          Credit card
+        </Text>
+        <Text as="p" variant="bodyMd">
+          Credit card information
+        </Text>
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/page-full-width.tsx]: Card doesn't apply spacing by default for children.
The below mimics the same styling as the previous LegacyCard section and aligns with our Storybook/doc examples.
```suggestion
        <BlockStack gap="200">
          <Text as="h2" variant="headingSm">
            Credit card
          </Text>
          <Text as="p" variant="bodyMd">
            Credit card information
          </Text>
        </BlockStack>
```

File: polaris.shopify.com/pages/examples/page-full-width.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import {Page, LegacyCard} from '@shopify/polaris';
-import {PlusIcon} from '@shopify/polaris-icons';
+import {Page, Card, Text} from '@shopify/polaris';
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/page-full-width.tsx]:
```suggestion
import {Page, Card, Text, BlockStack} from '@shopify/polaris';
```

To resolve possible CI errors with suggestion to fix Card spacing above.

File: polaris.shopify.com/pages/examples/page-narrow-width.tsx [hunk 4]
```diff
@@ -10,12 +11,17 @@ function PageExample() {
...
-      <LegacyCard title="Credit card" sectioned>
-        <p>Credit card information</p>
-      </LegacyCard>
+      <Card>
+        <Text as="h2" variant="headingSm">
+          Credit card
+        </Text>
+        <Text as="p" variant="bodyMd">
+          Credit card information
+        </Text>
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/page-narrow-width.tsx]: Same as above:
```suggestion
        <BlockStack gap="200">
          <Text as="h2" variant="headingSm">
            Credit card
          </Text>
          <Text as="p" variant="bodyMd">
            Credit card information
          </Text>
        </BlockStack>
```

File: polaris.shopify.com/pages/examples/page-narrow-width.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
-import {Page, LegacyCard, PageActions} from '@shopify/polaris';
+import {Page, PageActions, Card, Text} from '@shopify/polaris';
```
Comment (laurkim) [inline polaris.shopify.com/pages/examples/page-narrow-width.tsx]:
```suggestion
import {Page, PageActions, Card, Text, BlockStack} from '@shopify/polaris';
```

**Review by laurkim — APPROVED** [review]
Changes lgtm 👍 
Some minor suggestions to fix spacing on the replacement of LegacyCard with Card in the examples.

For visibility, UX/Eng on Polaris team approved suggestion of adding these examples in the Polaris Slack channel.

File: polaris-react/src/components/Page/components/Header/Header.tsx (hunk 1 above)
Comment (itwasmattgregg) [reply polaris-react/src/components/Page/components/Header/Header.tsx]: Moved

File: .changeset/silly-hairs-look.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+'polaris.shopify.com': minor
+---
+
+Adds examples for Card and Page with icon actions
```
Comment (laurkim) [inline .changeset/silly-hairs-look.md]:
```suggestion
Added examples for `Card` and `Page` with icon actions
```
Forgot to add this in earlier review but changesets are usually in past tense
